domains.js

```diff
@@ -0,0 +1,72 @@
+const google = require('googleapis');
+const sheets = google.sheets('v4');
+
+function getDomains(username) {
+    const credentials = credentialConfiguration();
+    const request = {
+        spreadsheetId: credentials.TASK_SHEET_ID,
+        range: 'A:ZZZ', // extreme range to get whole spreadsheet
+        auth: credentials.API_KEY
+    };
+
+    return new Promise((resolve, reject) => {
+        sheets.spreadsheets.values.get(request, function(err, response) {
+            if (err) {
+                reject(err);
+            } else {
+                const domains = findMatch(username, response.values);
+                if (domains) {
+                    resolve({ 'domains': domains })
+                } else {
+                    reject({ 'error': `${username} not found.`})
+                }
+            };
+        });
+    });
+}
+
+function findMatch(username, records) {
+    let domains = records.filter(record => {
+        return username.toLowerCase() === record[0].toLowerCase()
+    })[0];
+
+    if (domains) {
+        return domains.slice(1);
+    } else {
+        return null;
+    }
+}
+
+let baseEnvironment;
+
+/**
+ * Create a credential object with secret values from process.env.
+ * In development, also read from a file named .env, if present.
+ * @returns {Object}
+ */
+function credentialConfiguration () {
```

Comment:
Half this is the same as the `clientConfiguration()` function in `app.js`, so we should probably abstract it out into its own config module that grabs the combined ENV vars and `.env` file content. Then the `clientConfiguration()` function can call that and do its additional filtering.

For the server-side config (e.g. `TASK_SHEET_ID` and `API_KEY`), I don’t think we actually need to filter the values down like you are doing here. We only filter the values for the client config because we don’t want secret credentials to be made public through the API. Stuff that operates entirely within the server (like this domains module) doesn’t need to worry about that.